src/core.c/CompUnit/RepositoryRegistry.pm6

```diff
@@ -302,15 +302,53 @@ class CompUnit::RepositoryRegistry {
         Nil
     }
 
+    # Try to locate a distribution by a given file name. Only makes sense for CURFS.
+    # $file is expected to be either absolute or relative to $*CWD.
+    method distribution-for-file(::?CLASS:U:
+      $file, :$name, :$ver, :$auth, :$api
+    --> CompUnit::Repository::Distribution:D) {
+        my @distros =
+            $*REPO.repo-chain.map({
+                # TODO This CATCH is better be removed when old ecosystem implementations of repositories are updated
+                # to support the new API. In particular these are: Raku::CompUnit::Repository::Tar,
+                # Raku::CompUnit::Repository::Lib, and Raku::CompUnit::Repository::Github.
+                CATCH {
+                    when X::Multi::NoMatch {
+                        .rethrow unless .dispatcher.name eq 'candidates';
+                    }
+                }
+                .?candidates(:file(.?normalize-path($file) // $file), :$name, :$auth, :$api).head
+            }).grep(*.defined);
+        +@distros
+            ?? (@distros == 1 ?? @distros !! @distros.sort(*.meta<ver>).sort(*.meta<api>).reverse).head
+            !! Nil
+    }
+
+    # Find all candidate distributions for a given spec
+    proto method candidates(|) {*}
```

Comment:
I'm still not exactly convinced this `candidates` method (or `distribution-for-file` even) belong in this class. `CompUnit::RepositoryRegistry` deals with repositories, and those repositories may or may not deal with distributions (and the ones that *do* deal with distributions may not deal with *raku* distributions). I understand this logic needs to live somewhere, but it just seems wrong to include a method that "Only makes sense for CURFS" in an object used for managing generic repositories. I imagine it could live somewhere else, and while it might make using it from whatever callsite require a bit more logic, it would be more cohesive imo.